### PR TITLE
Transactions

### PR DESCRIPTION
**Pull request by youmari**
## In this branch I did the following : 
- Update  and Delete records safely 
- Add species column to the table animals 
- Perform  transactions
- Add queries using aggregate functions and group by

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -15,3 +15,15 @@ SELECT * FROM animals WHERE neutered = 'true';
...
+
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) FROM animals;
+
+SELECT neutered, COUNT(escape_attempts) FROM animals GROUP BY neutered;
+
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```
Comment (mmsesay) [inline queries.sql:79]: - [x] You did well writing these queries. However, you need to implement transaction queries as well, just as stated in main bullet points 4 & 5 of the [project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md)

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @youmari , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (youmari) [reply queries.sql:79]: thanks for your feedback just a misconception happened

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,63 @@ SELECT * FROM animals WHERE neutered = 'true';
...
+
+BEGIN;
+
+ALTER TABLE animals RENAME COLUMN species TO unspecified;
+
+\d animals;
+
+ROLLBACK;
+
+\d animals;
```
Comment (DammyShittu) [inline queries.sql:27]: - Kindly `UPDATE` the `animals` table and `SET` the `species` equal to 'unspecified' as the requirement says. 😉

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @youmari, 

Great work on making the changes requested by a previous reviewer 👏
 
You've done well implementing some of the requested changes, but there are still a tiny part that hasn't been addressed yet.

 ## Required Changes ♻️

_Check the comment under the review._

## Optional suggestions

- I can see that you're currently hard coding the `id` of each record/row being added. I would suggest that you add the `GENERATED ALWAYS AS IDENTITY` constraint to the `id` column. This will help you to generate sequential integers that cannot be altered when the record is being created.
You can go through this [guide](https://www.sqltutorial.org/sql-identity/#:~:text=The%20GENERATED%20ALWAYS%20generates%20sequential,integers%20for%20the%20identity%20column.) for more details. 😉 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (youmari) [reply queries.sql:27]: @DammyShittu Hey that's what line 21 do it's update the species to unspecified

**Comment by youmari** [comment]
sorry by mistake I deleted the comment @DammyShittu

**Comment by DammyShittu** [comment]
> sorry by mistake I deleted the comment @DammyShittu

Hi @youmari,
The query on line 21 will change the column name from `species` to `unspecified`. However, that is not what was mentioned in the project requirement as seen in the screenshot above ☝️.

The `species` column for each row should read `unspecified` after executing the query and not change the column name to `unspecified`.

The correct query should look like this 👇 

```
UPDATE animals
   SET species = 'unspecified'; 
```

I hope this helps. 😃

**Comment by youmari** [comment]
okay I got it  thanks a lot

**Review by DammyShittu — APPROVED** [review]
Hi @youmari, 

## APPROVED 🟢 

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**